pkg/network/encoding/json.go

```diff
@@ -21,12 +22,16 @@ type jsonSerializer struct {
 	marshaller jsonpb.Marshaler
 }
 
-func (j jsonSerializer) Marshal(conns *network.Connections) ([]byte, error) {
-	payload := modelConnections(conns)
-	writer := new(bytes.Buffer)
-	err := j.marshaller.Marshal(writer, payload)
-	returnToPool(payload)
-	return writer.Bytes(), err
+func (j jsonSerializer) Marshal(conns *network.Connections, writer io.Writer) error {
+	out := bytes.NewBuffer(nil)
+	modelConnections(model.NewConnectionsBuilder(out), conns)
+
+	var payload model.Connections
+	if err := payload.Unmarshal(out.Bytes()); err != nil {
+		return err
+	}
+
+	return j.marshaller.Marshal(writer, &payload)
```

Comment:
gostreamer does not have a way to emit json, so the way to get json is to go `network.Connections -> proto bytes -> model.Connection -> json (via jsonpb)`

Given this endpoint is only used for debugging purposes, I wasn't too concerned about the performance implications.

Reply:
While that's true, I think we should still measure the impact of the new implementation on the debug endpoint, as for example if it suddenly consumes 100MB, it might lead to OOMs (staging/production/customer) and lost of precious debugging information